Packs/McAfee_Advanced_Threat_Defense/Integrations/McAfee_Advanced_Threat_Defense/McAfee_Advanced_Threat_Defense.py

```diff
@@ -409,59 +413,105 @@ def url_upload_raw(body):
                 skip_task_id=None, analyze_again=None, x_mode=None, message_id=None,
-                file_priority_q=None, src_ip=None, dest_ip=None, file_name=None):
+                file_priority_q=None, src_ip=None, dest_ip=None, file_name=None, given_url=None):
     body = {}  # type: dict
     body['data'] = {}
     data = {}  # type: dict
     data['data'] = {}
     # Add missing prefix to url
-    if submit_type != 0:
-        if not sample.startswith('http://') and not sample.startswith('https://'):
-            if sample.startswith('www.'):
-                sample = "http://" + sample
-            else:
-                sample = "http://www." + sample  # disable-secrets-detection
+    if submit_type in SUBMIT_TYPE_WITH_URL:
+        sample = add_prefix_to_given_url(sample)
+    elif submit_type == 2:
+        given_url = add_prefix_to_given_url(given_url)
 
     data['data']['vmProfileList'] = vm_profile_list
     data['data']['submitType'] = submit_type
     data['data']['messageId'] = message_id
     data['data']['srcIp'] = src_ip
     data['data']['destIp'] = dest_ip
-    data['data']['url'] = '' if submit_type == 0 else sample
+    data['data']['url'] = get_url_entry_by_submit_type(submit_type, given_url, sample)
     data['data']['skipTaskId'] = int(skip_task_id) if skip_task_id else None
     data['data']['analyzeAgain'] = analyze_again
     data['data']['xMode'] = x_mode
     data['data']['filePriorityQ'] = file_priority_q if file_priority_q else 'run_now'
 
     body['data'] = json.dumps(data)
-    file_entry_id = sample if submit_type == 0 else ''
-    filename_to_upload = file_name if (submit_type == 0 and file_name) else ''
-    if submit_type == 0:
+    file_entry_id = sample if submit_type in SUBMIT_TYPE_WITH_FILE else ''
+    filename_to_upload = file_name if (submit_type in SUBMIT_TYPE_WITH_FILE and file_name) else ''
+    if submit_type in SUBMIT_TYPE_WITH_FILE:
         result_obj = file_upload_raw(body, file_entry_id, filename_to_upload)
-    elif submit_type == 1:
+    elif submit_type in SUBMIT_TYPE_WITH_URL:
         result_obj = url_upload_raw(body)
     return {
         'taskId': result_obj['results'][0]['taskId'],
         'resultObj': result_obj
     }
 
 
-def file_upload_command():
-    args = demisto.args()
+def get_url_entry_by_submit_type(submit_type, given_url, sample):
+    """
+        Args:
+            submit_type (int) : SubmitType argument (can be one of those: 0,1,2,3)
+            given_url (str) : a url if SubmitType arg is 2, None otherwise.
+            sample (str) : a url (if SubmitType is 1 or 3 ) or a file entry id (if SubmitType is 0 or 2)
 
-    if ('entryID' in args and 'url' in args) or ('entryID' not in args and 'url' not in args):
+        Returns:
+            url entry value (str)
+
+    """
+    if submit_type == 0:
+        return ''
+    elif submit_type == 2:
+        return given_url
+    else:
+        return sample
+
+
+def handling_errors(args):
```

Comment:
Choose a better name for the function - it is only related to the fule upload command.
Add docstring and UT also.

Add a short comment before each if, so it would be easier to understand what's going on in it

Reply:
Done